src/main/zapHomeFiles/hud/drawer.js

```diff
@@ -12,19 +12,57 @@ Vue.component('history', {
     template: '#history-template',
     data() {
         return {
-            messages: []
+            filter: '',
+            regexEnabled: false,
+            messages: [],
+            hiddenMessageCount: 0
         }
     },
     computed: {
         timeDescendingMessages() {
             return this.messages.slice().reverse();
+        },
+        messageCount() {
+            return this.messages.length;
+        },
+        filteredMessages() {
+            const self=this,
+                  isRegex = this.regexEnabled;
+            let re;
+
+            if (isRegex){
+                try {
+                    re = new RegExp(this.filter);
+                }
+                catch (ex) {
+                    re = new RegExp(''); //hack for now to prevent errors being thrown for invalid RegExp
```

Comment:
what would be the "non hack" solution? would it be to also alert the user of invalid regex?

Reply:
I was thinking the "non hack" solution would be to add some visual indicator to the user that the current RegEx was invalid, like a red outline around the input box? Would be easy to wire up to a state in this `catch` as they type out and can easily understand why no messages are showing. Kind of thinking how wireshark changes from green background to red as your typing in filters but more subtle. What you think?

I could do it on this PR, was really trying to get this in before launch but since I missed that I don't mind taking a little more time for polish.

Reply:
I think that's a great idea! if you can add that in without too much more effort I think that'd be a great addition. :)

Reply:
Implemented in latest revisions!